validation/src/main/scala/hmda/validation/engine/ts/TsValidationEngine.scala

```diff
@@ -1,59 +1,57 @@
 package hmda.validation.engine.ts
 
 import hmda.model.fi.ts.TransmittalSheet
-import hmda.validation.api.TsValidationApi
-import hmda.validation.dsl.{ Failure, Result, Success }
+import hmda.validation.api.ValidationApi
+import hmda.validation.api.ts.TsValidationApi
 import hmda.validation.engine.ValidationError
 import hmda.validation.rules.ts.syntactical._
-
 import scala.concurrent.{ ExecutionContext, Future }
-import scalaz.Scalaz._
 import scalaz._
+import scalaz.Scalaz._
 
-trait TsValidationEngine extends TsValidationApi {
+trait TsValidationEngine extends ValidationApi with TsValidationApi {
```

Comment:
This could be a long topic, and there is no one single answer to rule them all, but as a general rule of thumb I tend to think of these in the following way:

* Objects: these are singletons, which means one instance for that particular class. They are static, don't use the `new` keyword to get instantiated. They are the most basic, and are useful to encapsulate data and some functions. 

* Class. This is like a regular Java class (simplifying but it will do for now). You can have many instances of this class, they get instantiated with the `new` keyword and are used a lot if you are doing OO programming with Scala, a bit less if it's more functional (but they are still there; usually the entry point for a program, or some high level abstraction). There are abstract classes (need to be inherited, as in Java), and `case classes`, which are immutable and are one of the nicer features of Scala

* Traits: basically an "interface", with the particular feature that they allow multiple inheritance (unlike say, Java), and can also have an implementation (not just the function signature, but actual implementation that is inherited).

So, when to use which? Well, I guess it depends. Many things can be modelled with objects, specially if they are just data wrappers. The fact that `TsValidationEngine` is a trait is to give it more flexibility. You can see how the test (a ScalaTest class) extends it to have access to its functions and do testing, but that implementation is different from what will end up being the real engine, which will be based on `Akka` actors. Having traits in functions this way is very flexible for testing the business logic. In addition to this, you can start to see how the validation engine as a whole will be "composed" of several of these traits, that we can implement in a bit of isolation if needed. The division is a bit artificial right now (Transmittal Sheet vs LAR, syntactical, validity, etc.) but it's the way the domain is structured right now. Hopefully this rambling wasn't too long and this helped, but let me know otherwise.